src/dotnet/APIView/APIViewWeb/HostedServices/PullRequestBackgroundHostedService.cs

```diff
@@ -48,8 +52,19 @@ private async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             do
             {
-                await _pullRequestManager.CleanupPullRequestData();
-                await Task.Delay(6 * 60 * 60000, stoppingToken); //6 hours delay
+                try
+                {
+                    await _pullRequestManager.CleanupPullRequestData();
+                }
+                catch (Exception ex)
+                {
+                    _telemetryClient.TrackException(ex);
```

Comment:
Does the current behavior rely on this exception bubbling up at all right now? My guess is no just double checking since that's a functional change here.

Reply:
Nope. We are fine to handle the failure here and report to telemetry.